tests/integration/galley/analyze_test.go

```diff
@@ -180,6 +183,26 @@ func TestJsonInputFile(t *testing.T) {
 		})
 }
 
+func TestJsonOutput(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			g := NewGomegaWithT(t)
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-analyze",
+				Inject: true,
+			})
+
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+
+			// Validation error if we have a gateway with invalid selector.
+			output, err := istioctlSafe(t, istioCtl, ns.Name(), false, jsonGatewayFile, jsonOutput)
```

Comment:
Let's test a few other cases so we catch regressions in the future:
- no istio-system in cluster gives json with "2020-06-09T15:16:43.491907Z	warn	analysis	istio-system namespace not found. Istio may not be installed in the target cluster. Using default mesh configuration values for analysis"
- running on the istio examples gives json with "Skipping file sleep/README.md, recognized file extensions are: [.json .yaml .yml]"

Reply:
Thanks Prachi! I added a test for the invalid file input. Regarding the first warning you mentioned, our current tests are setup with Istio installed, IMO it is not worth it to uninstall istio and test that scenario. Is it possible to add a test in kpt for such scenario?